scipy/signal/signaltools.py

```diff
@@ -163,28 +162,28 @@ def correlate(in1, in2, mode='full'):
     elif not in1.ndim == in2.ndim:
         raise ValueError("in1 and in2 should have the same dimensionality")
 
-    if mode == 'valid':
-        _check_valid_mode_shapes(in1.shape, in2.shape)
-        # numpy is significantly faster for 1d
-        if in1.ndim == 1 and in2.ndim == 1:
-            return np.correlate(in1, in2, mode)
+    # numpy is significantly faster for 1d
+    if in1.ndim == 1 and in2.ndim == 1 and (in1.size >= in2.size):
```

Comment:
In `scipy/signal/tests/test_signaltools.py`:
- `TestCorrelateComplex64.test_swap_same`
- `TestConvolve.test_swap_same`
- `TestCorrelateComplex128.test_swap_same`
- `TestCorrelateComplex256.test_swap_same`

I suspect the issue is that `np.correlate` doesn't quite behave in the way `scipy` is intending for the `same` mode.  Now the question might be: "why not add a condition to check for `mode == same`" to which I would respond: did we really simplify anything then?